xdsl/dialects/tensor.py

```diff
@@ -683,6 +683,33 @@ class FromElementsOp(IRDLOperation):
     result = result_def(TensorType.constr(ELEMENT_TYPE))
     assembly_format = "$elements attr-dict `:` type($result)"
 
+    traits = traits_def(NoMemoryEffect())
+
+    def __init__(
+        self,
+        elements: Sequence[SSAValue] | SSAValue,
+    ):
+        if isinstance(elements, SSAValue):
+            shape: tuple[int, ...] = tuple()
+            elem_type = elements.type
+            operands = [elements]
+        else:
+            shape = (len(elements),)
+            if len(elements) > 0:
+                assert all(elem.type == elements[0].type for elem in elements)
```

Comment:
This verification is done by the verifier synthesized by the irdl annotation already, so no need to check for it here as well